tools/debug/transplanter/main.go

```diff
@@ -0,0 +1,484 @@
+			return
+		}
+		if n != int(lenMsg) {
+			output <- txGroupItem{err: fmt.Errorf("incomplete message body in %s", filePath)}
+			return
+		}
+
+		dec := codec.NewDecoderBytes(msgBuff, new(codec.MsgpackHandle))
+		var txgroup []transactions.SignedTxn
+		for {
+			var stx transactions.SignedTxn
+			err := dec.Decode(&stx)
+			if err == io.EOF {
+				break
+			} else if err != nil {
+				output <- txGroupItem{err: err}
+				return
+			}
+			txgroup = append(txgroup, stx)
+		}
+		output <- txGroupItem{ts: headers.Ts, txgroup: txgroup, path: filePath}
+	}
+}
+
+func readTransactions(output chan txGroupItem) {
+	defer close(output)
+	if len(*txnFile) > 0 {
+		data, err := os.ReadFile(*txnFile)
+		if err != nil {
+			err = fmt.Errorf("cannot read transaction file %s: %v", *txnFile, err)
+			output <- txGroupItem{err: err}
+			return
+		}
+
+		txgroup, err := decodeTxGroup(data)
+		if err != nil {
+			txgroup, err = decodeTxGroupSlices(data)
+			if err != nil {
+				err = fmt.Errorf("cannot decode transaction file %s: %v", *txnFile, err)
+				output <- txGroupItem{err: err}
+				return
+			}
+		}
+		output <- txGroupItem{ts: time.Time{}, path: *txnFile, txgroup: txgroup}
+	} else if len(*txnDir) > 0 {
+		files, err := os.ReadDir(*txnDir)
+		if err != nil {
+			err = fmt.Errorf("cannot read transaction directory %s: %v", *txnDir, err)
+			output <- txGroupItem{err: err}
+			return
+		}
+		for _, file := range files {
+			if file.IsDir() {
+				continue
+			}
+			path := filepath.Join(*txnDir, file.Name())
+			data, err := os.ReadFile(path)
+			if err != nil {
+				err = fmt.Errorf("cannot read transaction file %s: %v", path, err)
+				output <- txGroupItem{err: err}
+				return
+			}
+
+			txgroup, err := decodeTxGroup(data)
+			if err != nil {
+				err = fmt.Errorf("cannot decode transaction file %s: %v", path, err)
+				output <- txGroupItem{err: err}
+				return
+			}
+
+			output <- txGroupItem{ts: time.Time{}, path: path, txgroup: txgroup}
+		}
+	} else {
+		if len(*txnLogDir) > 0 {
+			files, err := os.ReadDir(*txnLogDir)
+			if err != nil {
+				err = fmt.Errorf("cannot read transaction log directory %s: %v", *txnLogDir, err)
+				output <- txGroupItem{err: err}
+				return
+			}
+			for _, file := range files {
+				if file.IsDir() {
+					continue
+				}
+				path := filepath.Join(*txnLogDir, file.Name())
+				var wg sync.WaitGroup
+				wg.Add(1)
+				go transcribeSnappyLog(path, output, &wg)
+				wg.Wait()
+			}
+		} else {
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go transcribeSnappyLog(*txnLogFile, output, &wg)
+			wg.Wait()
+		}
+	}
+}
+
+func main() {
+	flag.Parse()
+
+	if *help || *helpShort || len(*dataDir) == 0 || *roundStart == 0 {
+		usage()
+		os.Exit(1)
+	}
+
+	if len(*txnDir) > 0 && len(*txnLogDir) > 0 {
+		fmt.Fprintln(os.Stderr, "Cannot specify both transactions and transaction logs dirs")
+		os.Exit(1)
+	}
+
+	if !*force {
+		fmt.Println("Running this command could damage your node installation, proceed anyway (N/y)?")
```

Comment:
What is actually at risk of being broken by running this tool?

Reply:
yon cannot reuse this node dir after such a custom block committed. I think I need to change the message ask ensure there is a backup copy. For example, catching up mainnet might take hours, and can be easily spoiled by running this tool. Not a big deal for one and a problem for others (say archival node dirs).